src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1591,9 +1591,15 @@ static int client_parse_message(
         case DHCP_STATE_REQUESTING:
         case DHCP_STATE_RENEWING:
         case DHCP_STATE_REBINDING:
-                if (r == DHCP_NAK)
+                if (r == DHCP_NAK) {
+                        if ((client->lease != NULL) &&
+                            (client->lease->server_address != lease->server_address))
```

Comment:
Please drop redundant `()`. Also, we allow to implicit cast pointers to boolean. So, 
```c
if (client->lease && client->lease->server_address == lease->server_address)
        return log_dhcp_client_errno(...);
```

Reply:
Thanks, changed.